### PR TITLE
00060 implement backend for a to do list Pt. 1

### PR DESCRIPTION
**Pull request by mtracewicz**
Because of the amount of changes required for this ticket I have decided to split it in two for the ease of reviewing.

This first part contains a bump to .net8, splitting backend into projects, introduction of a bruno collection and the initial implementation for todo list.

For now the supported operations are:
- Create
- Rename
- Delete
- Get
- List (there is no filtering and no sorting)

### CODE REVIEW

File: backend/src/api/Controllers/TodoController.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            var list = await _service.CreateList(user, name);
+            return Created(HttpContext.Request.Path.Add(new PathString($"/{list.Id}")), list);
+        }
+    }
+
+    [HttpPut("lists/{id}")]
+    public async Task<IActionResult> RenameList([FromRoute] int id, [FromBody] Request request)
+    {
+        var userId = Request.UserId();
+        _logger.LogDebug("User: {user} created: {list}", userId, request.Name);
```
Comment (Sojusan) [inline backend/src/api/Controllers/TodoController.cs:84]: The same log message as for the create, shouldn't it be something like "updated"?

**Review by Sojusan — CHANGES_REQUESTED** [review]
Some minor things to consider.

File: backend/bruno/KSummarized/ToDo/List/Create.bru [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  seq: 2
+}
+
+post {
+  url: https://localhost:5000/api/todo/lists?name=demo
+  body: none
+  auth: bearer
+}
+
+params:query {
```
[the remark this answers is not in the document]
Comment (mtracewicz) [reply backend/bruno/KSummarized/ToDo/List/Create.bru:13]: You are right I have done query param for quick testing,a and forgot to change this. You can see the desired approach at rename.

**Review by Sojusan — APPROVED** [review]
LGTM

**Pull request merged**